scripts/main/contextMenu.js

```diff
@@ -181,71 +196,88 @@ contextMenu.albumMulti = function (albumIDs, e) {
 	basicContext.show(items, e.originalEvent, contextMenu.close);
 };
 
-contextMenu.buildList = function (lists, exclude, action, parent = 0, layer = 0) {
-	const find = function (excl, id) {
-		for (let i = 0; i < excl.length; i++) {
-			if (parseInt(excl[i], 10) === parseInt(id, 10)) return true;
-		}
-		return false;
-	};
+/**
+ * @callback ContextMenuActionCB
+ *
+ * @param {(Photo|Album)} entity
+ */
 
+/**
+ * @callback ContextMenuEventCB
+ *
+ * @param {jQuery.Event} [e]
+ * @returns {void}
+ */
+
+/**
+ * @param {(Photo|Album)[]} lists
+ * @param {string[]} exclude list of IDs to exclude
+ * @param {ContextMenuActionCB} action
+ * @param {?string} [parentID=null] parentID
+ * @param {number} [layer=0]
+ *
+ * @returns {{title: string, disabled: boolean, fn: ContextMenuEventCB}[]}
+ */
+contextMenu.buildList = function (lists, exclude, action, parentID = null, layer = 0) {
 	let items = [];
 
-	let i = 0;
-	while (i < lists.length) {
-		if ((layer === 0 && !lists[i].parent_id) || lists[i].parent_id === parent) {
-			let item = lists[i];
+	lists.forEach(function (item) {
+		if ((layer !== 0 || item.parent_id) && item.parent_id !== parentID) return;
 
-			let thumb = "img/no_cover.svg";
-			if (item.thumb && item.thumb.thumb) {
-				if (item.thumb.thumb === "uploads/thumb/") {
-					if (item.thumb.type && item.thumb.type.indexOf("video") > -1) {
-						thumb = "img/play-icon.png";
-					}
-				} else {
-					thumb = item.thumb.thumb;
+		let thumb = "img/no_cover.svg";
+		if (item.thumb && item.thumb.thumb) {
+			if (item.thumb.thumb === "uploads/thumb/") {
+				if (item.thumb.type && item.thumb.type.indexOf("video") > -1) {
+					thumb = "img/play-icon.png";
 				}
-			} else if (item.sizeVariants) {
-				if (item.sizeVariants.thumb === null) {
-					if (item.type && item.type.indexOf("video") > -1) {
-						thumb = "img/play-icon.png";
-					}
-				} else {
-					thumb = item.sizeVariants.thumb.url;
+			} else {
+				thumb = item.thumb.thumb;
+			}
+		} else if (item.size_variants) {
+			// TODO: My IDE complains that thumb is always non-null, and I would agree
```

Comment:
As before, I think this is meant to handle cases of videos with missing thumbnails.